test/test_autograd.py

```diff
@@ -5635,6 +5635,11 @@ def test_leaky_relu_inplace_with_neg_slope(self, device):
         with self.assertRaisesRegex(RuntimeError, "call out-of-place version"):
             b.backward(torch.ones(2, device=device))
 
+        a = torch.tensor([-2., 0., 2.], device=device, requires_grad=True)
```

Comment:
kind of a strange test?  Like, I don't think you'd put these two tests together unless you followed this exact path to get to this state.  Otherwise you'd split up error cases and non-error cases.

On the non error cases: do we not test inplace functions through the normal autograd test path?  I don't see why we need a one-off test here that memorizes the output instead of normal gradchecks.

Reply:
This test is not for the specific inplace issue.
But the check we have is based on the assumption that we return a subgradient of 0 at 0. So this test is here to make sure that this assumption is true.

You think we should add a test to make sure we don't raise an error for inplace in the 0 case?

Reply:
@albanD Talked to @gchanan , I am going to put slope == 0.0 test into common_nn.py as a standard nn module test.